snowflake_blueprints/upload_file.py

```diff
@@ -100,14 +100,27 @@ def combine_folder_and_file_name(folder_name, file_name):
 
 def upload_data(source_full_path, table_name, insert_method, db_connection):
     try:
-        for chunk in pd.read_csv(source_full_path, chunksize=10000):
-            chunk.to_sql(
-                table_name,
-                con=db_connection,
-                index=False,
-                if_exists=insert_method,
-                method='multi',
-                chunksize=1000)
+        for index, chunk in enumerate(
+                pd.read_csv(source_full_path, chunksize=10000)):
+
+            if insert_method == 'replace' and index > 0:
+                # First chunk replaces the table, the following chunks
+                # append to the end.
+                chunk.to_sql(
+                    table_name,
+                    con=db_connection,
+                    index=False,
+                    if_exists='append',
+                    method='multi',
+                    chunksize=10000)
+            else:
+                chunk.to_sql(
+                    table_name,
+                    con=db_connection,
+                    index=False,
+                    if_exists=insert_method,
+                    method='multi',
+                    chunksize=10000)
```

Comment:
```
if insert_method == 'replace' and index > 0:
  insert_method = 'append'

chunk.to_sql(..., if_exists=insert_method)
```
Provides the same functionality without chunk.to_sql duplication.

Reply:
👍 I'll change that across the board.